test-suite/src/loan/tests/problem_with_approval_length_example.rs

```diff
@@ -0,0 +1,268 @@
+                                royalty_info: None,
+                            },
+                        },
+                    },
+                },
+                &[Coin {
+                    denom: NATIVE_DENOM.to_string(),
+                    amount: Uint128::new(100000u128),
+                }],
+            );
+
+            // VENDING_MINTER is minter
+            let mint_nft_tokens = app
+                .execute_contract(
+                    Addr::unchecked(OWNER_ADDR),
+                    Addr::unchecked(VENDING_MINTER),
+                    &vending_minter::msg::ExecuteMsg::Mint {},
+                    &[Coin {
+                        denom: NATIVE_DENOM.to_string(),
+                        amount: Uint128::new(100000u128),
+                    }],
+                );
+            assert!(mint_nft_tokens.is_ok());
+
+            // token id 41
+            let grant_approval = app
+                .execute_contract(
+                    Addr::unchecked(OWNER_ADDR),
+                    Addr::unchecked(SG721_CONTRACT),
+                    &sg721_base::msg::ExecuteMsg::<Empty, Empty>::Approve {
+                        spender: nft_loan_addr.to_string(),
+                        token_id: "41".to_string(),
+                        expires: Some(Expiration::AtHeight(current_block + 1)),
+                    },
+                    &[],
+                );
+            assert!(grant_approval.is_ok());
+
+            // good deposit single collateral
+            let good_deposit_collateral = app
+                .execute_contract(
+                    Addr::unchecked(OWNER_ADDR),
+                    nft_loan_addr.clone(),
+                    &ExecuteMsg::DepositCollaterals {
+                        tokens: vec![AssetInfo::Sg721Token(Sg721Token {
+                            address: SG721_CONTRACT.to_string(),
+                            token_id: "41".to_string(),
+                        })],
+                        terms: Some(LoanTerms {
+                            principle: Coin {
+                                denom: NATIVE_DENOM.to_string(),
+                                amount: Uint128::new(100),
+                            },
+                            interest: Uint128::new(50),
+                            // duration > approval
+                            duration_in_blocks: 15,
+                        }),
+                        comment: Some("be water, my friend".to_string()),
+                        loan_preview: None,
+                    },
+                    &[Coin {
+                        denom: NATIVE_DENOM.to_string(),
+                        amount: Uint128::new(100u128),
+                    }],
+                );
+            assert!(good_deposit_collateral.is_ok());
+
+            // then someone will accept the offer but approval is past due
+            let chainid = app.block_info().chain_id.clone();
+            app.set_block(BlockInfo {
```

Comment:
thank you for including the example logic to update the mock_apps block height & time!